lib/src/client.dart

```diff
@@ -140,3 +141,48 @@ abstract class Client {
   /// do so can cause the Dart process to hang.
   void close();
 }
+
+const _clientToken = Object();
+
+/// Return the [Client] for the current [Zone], if one has been set.
+///
+/// NOTE: This function is explicitly hidden from the public API.
+Client? get zoneClient => Zone.current[_clientToken] != null
+    ? Zone.current[_clientToken] as Client
+    : null;
+
+/// Runs [body] in its own [Zone] with [client] set as the default [Client].
+///
+/// For example:
+///
+/// ```
+/// class MyHttpClient extends http.BaseClient {
+///   @override
+///   Future<http.StreamedResponse> send(http.BaseRequest request) {
+///     // your implementation here
+///   }
+/// }
+///
+/// void main() {
+///  late Client client;
+///  if (Platform.isAndroid) {
+///    client = MyHttpClient();
+///  } else {
+///    client = Client();  // Use the default client.
+///  }
+///  runWithClient(myFunction, () => client);
+/// }
+///
+/// myFunction() {
+///   // Uses the `Client` configured in `main`.
+///   final response = await get(Uri.parse("https://www.example.com/"));
+///   final client = Client();
+/// }
+/// ```
+///
+/// The [client] influences functions (e.g. `get`) and the [Client.new] factory.
```

Comment:
"Influences" is too vague. Could mean anything.
```dart
/// The [Client] created by [client] is returned by the [Client.new] 
/// factory constructor, and is used by functions like [get].
```
(Don't know where `get` is defined, but make sure to link to it.)

Reply:
Done.

Reply:
> "Influences" is too vague. Could mean anything.

Done.